lib/logstash/inputs/http.rb

```diff
@@ -122,14 +122,20 @@ def register
     @server.min_threads = 0
     @server.max_threads = @threads
     @codecs = Hash.new
+
     @additional_codecs.each do |content_type, codec|
       @codecs[content_type] = LogStash::Plugin.lookup("codec", codec).new
     end
-    @codec_lock = java.util.concurrent.locks.ReentrantLock.new
+
+    num_write_slots = @threads > 2 ? @threads - 1 : 1
```

Comment:
Alternatively, you could do is start the web server with `threads + 1`  to allow `threads` number of write slots, and the extra one for reporting errors, but you've already increased performance...